packages/components/src/range-control/index.js

```diff
@@ -256,16 +274,19 @@ function RangeControl(
 					</AfterIconWrapper>
 				) }
 				{ withInputField && (
-					<InputField
+					<InputNumber
+						aria-label={ label }
+						className="components-range-control__number"
 						disabled={ disabled }
+						inputMode="decimal"
 						isShiftStepEnabled={ isShiftStepEnabled }
-						label={ label }
 						max={ max }
 						min={ min }
+						onBlur={ handleOnInputNumberBlur }
 						onChange={ handleOnChange }
-						onReset={ handleOnReset }
 						shiftStep={ shiftStep }
 						step={ step }
+						type={ 'number' }
```

Comment:
@stokesman The `type` should be `number` by default for `NumberControl`. I don't think we need this :)

Reply:
Quite right, it's gone!